common/src/service/HttpClientHelper.cs

```diff
@@ -763,6 +751,28 @@ public static bool IsMappedToException(HttpResponseMessage message)
             Func<HttpResponseMessage, CancellationToken, Task> processResponseMessageAsync,
             IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> errorMappingOverrides,
             CancellationToken cancellationToken)
+        {
+            await ExecuteAsync(httpClient,
+                httpMethod,
+                requestUri,
+                modifyRequestMessageAsync,
+                isMappedToException,
+                processResponseMessageAsync,
+                errorMappingOverrides,
+                false,
+                cancellationToken).ConfigureAwait(false);
+        }
+
+        async Task ExecuteAsync(
+            HttpClient httpClient,
+            HttpMethod httpMethod,
+            Uri requestUri,
+            Func<HttpRequestMessage, CancellationToken, Task> modifyRequestMessageAsync,
+            Func<HttpResponseMessage, bool> isMappedToException,
+            Func<HttpResponseMessage, CancellationToken, Task> processResponseMessageAsync,
+            IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> errorMappingOverrides,
+            bool infinteClientTimeout,
+            CancellationToken cancellationToken)
```

Comment:
What is the difference in behavior between the operation timeout vs cancellation token? In the `ExecuteAsync` method above, the operation timeout passed is used to create a cancellation token source, which is then passed to the HTTP request: https://github.com/Azure/azure-iot-sdk-csharp/pull/1178/files#diff-4e06d203c3b4754233cdbe29cc0ccfc8L724